Chisel/CHLDuplicateDetection.m

```diff
@@ -0,0 +1,49 @@
+//
+//  CHLDuplicateDetection.m
+//  Chisel
+//
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#if __has_feature(objc_arc)
+#error CHLObjectDuplicateDetection.m expects ARC to be disabled.
+#endif
+
+#import "CHLDuplicateDetection.h"
+
+#import "CHLObjectAllocations.h"
+
+#import <objc/runtime.h>
+
+NSArray *CHLFindDuplicates(Class cls, BOOL (^equalFunction)(id left, id right)) {
+  NSMutableArray *objects = [NSMutableArray new];
+  
+  CHLEnumerateObjectsWithBlock(^(id object) {
+    if (object_getClass(object) == cls) {
```

Comment:
I'm fine with both. I think using `isKindOfClass` for now might actually be better, since we might want to look at subclasses right away. Let me change that.